docker/debug/Dockerfile

```diff
@@ -1,11 +1,16 @@
 ARG golang_image
 
 FROM $golang_image AS build
+ARG TARGETARCH
 ENV GOPATH /go
-RUN apk add --update --no-cache ca-certificates make git && \
-    go get github.com/go-delve/delve/cmd/dlv && \
-    cd /go/src/github.com/go-delve/delve && \
-    make install
+RUN apk add --update --no-cache ca-certificates make git
+RUN if [[ "$TARGETARCH" != "s390x" ]] ; then \
+        go get github.com/go-delve/delve/cmd/dlv && \
+        cd /go/src/github.com/go-delve/delve && \
+        make install; \
+    else \
+        touch /go/bin/dlv; \
```

Comment:
Wait, what? Can you add a comment here describing what's going on? Is dlv not supported at all for s390x? If so, we shouldn't be publishing this image for this arch.

Reply:
`go-delve` does not have s390x support yet. Since Dockerfile of `release` version images also requires the `debug_image` arg to be populated, we did this in `docker/debug/Dockerfile` to make a different version of `debugimg` for s390x and eliminate cross-script dependencies of `debug_image` arg values.

BTW, `debug` version images such as `all-in-one-debug` and `jaeger-agent-debug` are not currently built for s390x (we use `--platform="linux/amd64"` arg in `docker buildx` command for these images).

Reply:
We need a comment here then, possibly instructing our future selves which conditions should exist for us to remove this. For instance, "Once go-delve adds support for s390x (see issue #...) remove this entire conditional"

Reply:
Comment added. Thanks!